Makefile.include

```diff
@@ -292,8 +292,12 @@ ifeq ($(origin RIOT_VERSION), undefined)
   endif
 endif
 
+# Set module by prepending APPLICATION name with 'application_'.
+# It prevents conflict with application and modules with the same name.
+APPLICATION_MODULE ?= application_$(APPLICATION)
```

Comment:
non-blocking suggestion: maybe shorten this a bit and use `app_<name>` here. 

And we might consider to apply this scheme more broadly and have `drv_<driver name>`, `pkg_<package name>`, `sys_<library name>`, and maybe also `net_<something>` for drivers, packages, system, and network stuff respectively?

Reply:
I would not like to call it "app_" because I am currently trying to make a difference between "application" in the riot sense and "apps" in a more android style where you can have different "apps" in the same system. Also it introduce another name "apps" which is different.

Not sure its required for other archive outputs, here its really only because its a common conflicts case between an application/test name and the used library.
In any case it would be another PR for this.